container/src/container_state.rs

```diff
@@ -0,0 +1,187 @@
+use crate::{Error, Result};
+use serde::{Deserialize, Serialize};
+use std::fs;
+use std::fs::{File, OpenOptions};
+use std::io::Write;
+use std::path::PathBuf;
+
+const KAPS_ROOT_PATH: &str = "/var/run/kaps";
+const OCI_VERSION: &str = "0.2.0";
+const STATE_FILE: &str = "state.json";
+
+/// Container runtime status
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub enum Status {
+    #[serde(rename = "creating")]
+    Creating,
+    #[serde(rename = "created")]
+    Created,
+    #[serde(rename = "running")]
+    Running,
+    #[serde(rename = "stopped")]
+    Stopped,
+}
+
+impl Default for Status {
+    fn default() -> Self {
+        Status::Creating
+    }
+}
+
+/// Represent the state of the running container.
+#[derive(Serialize, Deserialize)]
+pub struct ContainerState {
+    id: String,
+    /// Path where the state of the container will be saved.
+    #[serde(skip)]
+    container_path: PathBuf,
+    /// OCI version.
+    oci_version: String,
```

Comment:
What is this field used for?

Reply:
The container_path field is used to avoid the reconstruction of the container state path each time it is needed.

The oci_version field was specified in the default minimal OCI runtime state specification but has no purpose at the moment.

Reply:
AFAIU `container_path` is only used when we save the state. And there we `join` it to the `state.json` path. So we do a potential buffer reallocation there anyway. What I'm saying is that keeping that `PathBuf` is a micro optimization and I'd prefer if we keep it out of the state structure for now. That makes `ContainerState` a fully serializable structure, which is a desirable pattern.